data_model/src/domain.rs

```diff
@@ -82,8 +92,9 @@ impl Ord for NewDomain {
 }
 
 impl NewDomain {
+    /// Create a [`NewDomain`], reserved for internal use.
     #[must_use]
-    fn new(id: <Domain as Identifiable>::Id) -> Self {
+    pub fn new(id: <Domain as Identifiable>::Id) -> Self {
```

Comment:
I'm trying to make a uniform API and you're breaking it with this change. It was my intention to hide away `NewXXX` structures so that users almost never needs to reference them in their application, i.e. `Domain::new` would produce `NewDomain` which would offer builder methods on itself giving the impression that they are always working with just `Domain`.

Reply:
I must admit I don't think I understood the intention behind the API. To me, if this is the case, then the genesis block should *not* contain any of the NewXXX structures. I distinctly remember you specifically adding them to the genesis block. 

If the old approach of generating Domain, rather than NewDomain in the genesis block is acceptable, I'd happily revert these changes. 

However, if we want to retain NewDomain inside the genesis block, I'm not sure that the new API is compatible with that.